WordPressFlux/WordPressFlux/Sources/QueryStore.swift

```diff
@@ -58,4 +113,62 @@ open class QueryStore<State, Query>: StatefulStore<State>, Unsubscribable {
     open func queriesChanged() {
         // Subclasses should implement this
     }
+
+}
+
+// In ideal world, those wouldn't be extensions on `Encodable`/`Codable`, but rather refinements of
+// `QueryStore` in which `Store` conforms to `Codable` — then we would just change behavior _inside_ the `QueryStore`,
+// based on whether the type conforms to `Codable` or not.
+// Regretfully, Swift type system (or my Swift skills) does't allow to express that —
+// you can't just do `let codableSelf = self as? QueryStore<Codable, Query>` because
+// _Codable itself doesn't conform to Codable_.
+private extension Encodable {
+    func storeCachedState(at url: URL) {
+        do {
+            let jsonEncoder = JSONEncoder.init()
+            let encodedStore = try jsonEncoder.encode(self)
+
+            try encodedStore.write(to: url, options: [.atomic])
+        } catch {
+            NSLog("[PluginStore Error] \(error)")
+        }
+    }
+}
+
+private extension Decodable {
+    static func cachedState(from cachedURL: URL) -> Self? {
+        do {
+            let data = try Data(contentsOf: cachedURL)
+            let state = try JSONDecoder().decode(Self.self, from: data)
+
+            return state
+        } catch {
+            NSLog("[PluginStore Error] \(error)")
+            return nil
+        }
+    }
+}
+
+private extension QueryStore {
+    private static func cacheFileURL() throws -> URL {
+        let cacheFilename =  "\(String(describing: self)).json"
+        let documentsPath = try FileManager.default.url(for: .cachesDirectory,
+                                                        in: .userDomainMask,
+                                                        appropriateFor: nil,
+                                                        create: true)
+
+        let targetURL = documentsPath.appendingPathComponent(cacheFilename)
+
+        return targetURL
+    }
+}
+
+extension QueryStore where State: Encodable {
+    public func storeCachedState() {
```

Comment:
The cache naming still doesn't feel completely right, I'd talk about persistence instead:

- `cacheFileURL` -> `persistenceURL`
- `storeCachedState` -> `persistState`

What do you think?